src/main/java/org/jabref/gui/push/PushToApplicationMenuAction.java

```diff
@@ -0,0 +1,61 @@
+package org.jabref.gui.push;
+
+import java.util.Optional;
+
+import javafx.scene.control.MenuItem;
+
+import org.jabref.Globals;
+import org.jabref.gui.actions.Action;
+import org.jabref.gui.actions.SimpleCommand;
+import org.jabref.gui.icon.JabRefIcon;
+import org.jabref.gui.keyboard.KeyBinding;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.preferences.JabRefPreferences;
+
+/**
+ * An Action class representing the process of invoking a PushToApplicationMenu operation.
+ */
+public class PushToApplicationMenuAction extends SimpleCommand {
+
+    private final PushToApplication application;
+    private final PushToApplicationAction pushToApplicationAction;
+    private MenuItem pushToApplicationMenuItem;
+
+    public PushToApplicationMenuAction(PushToApplication pushToApplication, PushToApplicationAction pushToApplicationAction, MenuItem pushToApplicationMenuItem) {
+        this.application = pushToApplication;
+        this.pushToApplicationAction = pushToApplicationAction;
+        this.pushToApplicationMenuItem = pushToApplicationMenuItem;
+    }
+
+    public Action getActionInformation() {
+        return new Action() {
+
+            @Override
+            public Optional<JabRefIcon> getIcon() {
+                return Optional.of(application.getIcon());
+            }
+
+            @Override
+            public Optional<KeyBinding> getKeyBinding() {
+                return Optional.empty();
+            }
+
+            @Override
+            public String getText() {
+                return Localization.lang(application.getApplicationName());
+            }
+
+            @Override
+            public String getDescription() {
+                return "";
+            }
+        };
+    }
+
+    @Override
+    public void execute() {
+        Globals.prefs.put(JabRefPreferences.PUSH_TO_APPLICATION, application.getApplicationName());
+        pushToApplicationAction.updateApplication(application);
+        //pushToApplicationMenuItem.setText(pushToApplicationAction.getActionInformation().getText());
```

Comment:
This does not work. Any ideas?

Reply:
Hm, do you get any error message?

Reply:
This code works fine so far, I just put this last line in comment to show what I'm going to do. When I compile "live" with this last line, JabRef displays a runtime error in "execution".

Reply:
Please post the stacktrace and the error message. Something like a bound value cannot be set?
That would imply that there is somehow a Binding on that property

Reply:
java.lang.RuntimeException: MenuItem.text : A bound value cannot be set.
	at javafx.beans.property.StringPropertyBase.set(StringPropertyBase.java:140)
	at javafx.beans.property.StringPropertyBase.set(StringPropertyBase.java:49)
	at javafx.scene.control.MenuItem.setText(MenuItem.java:257)
	at org.jabref.gui.push.PushToApplicationMenuAction.execute(PushToApplicationMenuAction.java:59)
	at org.jabref.gui.actions.JabRefAction.lambda$new$30(JabRefAction.java:29)
	at org.controlsfx.control.action.Action.handle(Action.java:419)
	at org.controlsfx.control.action.Action.handle(Action.java:64)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.control.MenuItem.fire(MenuItem.java:462)
	at com.sun.javafx.scene.control.skin.ContextMenuContent$MenuItemContainer.doSelect(ContextMenuContent.java:1405)
	at com.sun.javafx.scene.control.skin.ContextMenuContent$MenuItemContainer.lambda$createChildren$343(ContextMenuContent.java:1358)
	at com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.Scene$MouseHandler.process(Scene.java:3757)
	at javafx.scene.Scene$MouseHandler.access$1500(Scene.java:3485)
	at javafx.scene.Scene.impl_processMouseEvent(Scene.java:1762)
	at javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2494)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:394)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:295)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$358(GlassViewEventHandler.java:432)
	at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:410)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:431)
	at com.sun.glass.ui.View.handleMouseEvent(View.java:555)
	at com.sun.glass.ui.View.notifyMouse(View.java:937)
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at com.sun.glass.ui.win.WinApplication.lambda$null$152(WinApplication.java:177)
	at java.lang.Thread.run(Thread.java:748)

Reply:
As I expected. There is a binding on the MenuItem.textProperty() (Binding is javafx (bi)directional databinding on properties). I think this could be related to the PR from @tobiasdiez  in which he implemeted the disabling of menu items.

Reply:
Mh, I have no experience with bindings, but this sounds like, i should better not change something, which is still WiP by @tobiasdiez. As this is just eye-candy, in can make this label static and freeze also the symbol in the Toolbar. (I was surprised, it actually changes to the proper symbol, if you restart JabRef)

Reply:
Actually it's very practical: https://docs.oracle.com/javase/8/javafx/properties-binding-tutorial/binding.htm
And also see our wiki for some general javafx explainations/tips
https://github.com/JabRef/jabref/wiki/Code---JavaFX